File: B3835 [GESP202303 一级] 每月天数.cpp
#include <iostream>
using namespace std;

bool runnian(int year) {
    return (year % 4 == 0 && year % 100 != 0) || (year % 400 == 0);
}

int daymounter(int year, int month) {
    int daysinmonth[] = {0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    if (month == 2 && runnian(year)) {
        return 29;
    }
    return daysinmonth[month];
}

int main() {
    int year, month;
    cin >> year >> month;
    cout << daymounter(year, month) << endl;
    return 0;
}

File: B3835 [GESP202303 一级] 每月天数.py
def runnian(year):
    return (year%4==0 and year%100 !=0) or (year%400==0)
def daymounter(year,month):
    daysinmonth={
        1:31,
        2:29 if runnian(year) else 28,
        3:31,
        4:30,
        5:31,
        6: 30,
        7: 31,
        8: 31,
        9: 30,
        10: 31,
        11: 30,
        12: 31
    }
    return daysinmonth[month]
year, month = map(int, input().split())
#wozhendefulezhegekaifa
print(daymounter(year, month))
